src/sentry/incidents/logic.py

```diff
@@ -1452,3 +1453,50 @@ def translate_aggregate_field(aggregate, reverse=False):
                 if translated_field == column:
                     return aggregate.replace(column, field)
     return aggregate
+
+
+def get_slack_channel_ids(organization, user, data, peek=False):
+    # If peek is True, we do no actual lookups.
+    # We simply return True if there is a lookup to do. Otherwise {}.
+    try:
+        from sentry.incidents.endpoints.serializers import AlertRuleTriggerActionSerializer
+
+        mapped_ids = {}
+        for trigger in data["triggers"]:
+            for action in trigger["actions"]:
+                a_s = AlertRuleTriggerActionSerializer(
+                    context={
+                        "organization": organization,
+                        "access": SystemAccess(),
+                        "user": user,
+                    },
+                    data=action,
+                )
+                if a_s.is_valid():
```

Comment:
@iProgramStuff I should note that the way this function is structured, we will still might make API calls if one of the triggers is invalid. Maybe that's OK for now but it's not ideal.

Reply:
Not sure if I follow

Reply:
@iProgramStuff correct, but where is the validation error being raised?

Reply:
In the various validate functions in AlertRuleTriggerActionSerializer when we call `is_valid()`